### ISSUE TITLE
Add an ingorePath option to the `no-unsupported-features` rules

### ISSUE THREAD

**Issue by pvdlg** [mysticatea/eslint-plugin-node#97]
In some cases a project can write both code that will run on Node 4 and code that will be transpiled.

For example [ava](https://github.com/avajs/ava) allow to write test code in es2017.
So for example a project that targets Node 4 will:
- Have `	"engines": {"node": ">=4"}` in the package.json
- Sources files `./index.js`, `./src`, `./bin` that are supposed to written for Node 4. Those files must be checked.
- Test files in `./test` that will be transpiled by [ava](https://github.com/avajs/ava). Those must not be checked.

An `ignorePath` option would allow to configure the path to ignore (in this case `./test`).
In addition it might be a good idea to ignore by default all paths and files not included in the `npm` package (`files` properties in `package.json` + `.npmignore`).

**Comment by mysticatea** [comment]
Thank you for this issue.

I'm not sure that the option is needed because I think you can use `overrides` property of `.eslintrc.*` file. For example:

```js
{
    "plugins": ["node"],
    "extends": ["eslint:recommended", "plugin:node/recommended"],
    "rules": {
    },
    "overrides": [
        {
            "files": ["test/**/*.js"],
            "rules": {
                "node/no-unsupported-features": ["error", { "ignores": ["asyncAwait"] }]
            },
        }
    ]
}
```

**Comment by pvdlg** [comment]
Ok, it totally makes sense!

Thinking more about it, maybe it would be a good idea to add an `engines` option to the rule. The options would have the same format as the one in `package.json` and it will override the what's configured in `package.json`.

So in the example of a Node 4 project that uses `ava` we could configure:
- the rule with default behavior for the sources
- the rule with `{engines: node: '>=8'}` for the test files

The current `ignore` option is great and allow for very specific control, but it seems that most project do no set Babel transform one by one, but instead use an es version or a node version as a target.

This is especially true since [babel-preset-env](https://github.com/babel/babel-preset-env) is the recommended approach.

**Comment by mysticatea** [comment]
It has `version` option: https://github.com/mysticatea/eslint-plugin-node/blob/master/docs/rules/no-unsupported-features.md#version

**Comment by pvdlg** [comment]
I'm so sorry...I don't know how I missed that....

I'm gonna close the issue as it's irrelevant. Thanks for the help!
